src/core/interpreter.cc

```diff
@@ -805,9 +809,16 @@ int Interpreter::RedisGenericCommand(bool raise_error, bool async) {
    * and this way we guaranty we will have room on the stack for the result. */
   lua_pop(lua_, argc);
   RedisTranslator translator(lua_);
-  redis_func_(CallArgs{MutSliceSpan{args}, &buffer, &translator, async, raise_error, &raise_error});
+  redis_func_(
+      CallArgs{MutSliceSpan{args}, &buffer_, &translator, async, raise_error, &raise_error});
   cmd_depth_--;
 
+  // Shrink reusable buffer if it's too big.
+  if (buffer_.capacity() > 128) {
+    buffer_.clear();
+    buffer_.shrink_to_fit();
+  }
+
```

Comment:
If the script does something like
```
while true:
  if not call: EXISTS key
    call HSET *lots of data*
```

you'll get a grow-shrink look and will actually have an allocation every single time 

Though its still better than allocating a buffer each time